QWeb/internal/xhr.py

```
@@ -16,12 +16,256 @@
+        const dec = function(){ try { window.__xhrMon.pending--; } catch(e) {} };
+        return p && typeof p.finally === "function"
+          ? p.finally(dec)
+          : p.then(function(r){ dec(); return r; }, function(e){ dec(); throw e; });
+      };
+      window.__xhrMon.fetchPatched = true;
+    }
+  } catch(e){}
+
+  try {
+    if (!window.__xhrMon.xhrPatched && window.XMLHttpRequest && window.XMLHttpRequest.prototype) {
+      const _open = XMLHttpRequest.prototype.open;
+      const _send = XMLHttpRequest.prototype.send;
+      XMLHttpRequest.prototype.open = function() { this.__xhrTracked = true; return _open.apply(this, arguments); };
+      XMLHttpRequest.prototype.send = function() {
+        if (this.__xhrTracked) {
+          try { window.__xhrMon.pending++; } catch(e) {}
+          this.addEventListener("loadend", function(){ try { window.__xhrMon.pending--; } catch(e) {} }, { once:true });
+        }
+        return _send.apply(this, arguments);
+      };
+      window.__xhrMon.xhrPatched = true;
+    }
+  } catch(e){}
+
+  try {
+    if (!window.__xhrMon.observerStarted && window.MutationObserver) {
+      const obs = new MutationObserver(function(){ window.__xhrMon.lastMutationTs = Date.now(); });
+      obs.observe(document.documentElement || document.body, {
+        childList:true, subtree:true
+      });
+      window.__xhrMon.observerStarted = true;
+    }
+  } catch(e){}
+
+  return true;
+})();
+"""
+# This was modified above:
+# obs.observe(document.documentElement || document.body, {
+#        childList:true, subtree:true, attributes:true, characterData:true
+#      });
+
+# Lightweight status (no spinner logic here anymore)
+JS_STATUS_LITE = r"""
+return (function (quietMs) {
+  var ready = (document.readyState === "complete");
+
+  var jqActive = null;
+  try {
+    if (typeof jQuery !== "undefined" && jQuery && typeof jQuery.active === "number") jqActive = jQuery.active;
+  } catch(e){}
```

Silent error discarding of all errors seems bit questionable.